### PR TITLE
Add section for new contributors

### PR DESCRIPTION
**Pull request by Gouthamve**
Inspiration from this blog post: https://manishearth.github.io/blog/2016/01/03/making-your-open-source-project-newcomer-friendly/

Some parts are lifted from: https://github.com/servo/servo/blob/master/CONTRIBUTING.md

---
This change is [Reviewable](https://reviewable.io/reviews/prometheus/prometheus/2980)

### CODE REVIEW

**Comment by Gouthamve** [comment]
Added the labels info here: https://github.com/prometheus/prometheus/wiki/Label-Names-and-Descriptions

Let me know if the location is okay or not. I will link it in the file and update the PR if its okay.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -16,3 +18,34 @@ Prometheus uses GitHub to manage reviews of pull requests.
...
+make test         # For more thorough testing before you commit and push.
+```
+
+## Pull Request Checklist
+
+* Branch from the master branch and, if needed, rebase to the current master branch before submitting your pull request. If it doesn't merge cleanly with master you may be asked to rebase your changes.
+
+* Commits should be as small as possible, while ensuring that each commit is correct independently (i.e., each commit should compile and pass tests).
+
+* If your patch is not getting reviewed or you need a specific person to review it, you can @-reply a reviewer asking for a review in the pull request or a comment, or you can ask for a review on IRC channel [#prometheus](https://webchat.freenode.net/?channels=#prometheus) on irc.freenode.net (for the easiest start, [join via Riot](https://riot.im/app/#/room/#prometheus:matrix.org)).
```
Comment (brian-brazil) [inline CONTRIBUTING.md]: We probably shouldn't duplicate the community page here. Links like these have a tendency to rot over the years.

Comment (Gouthamve) [inline CONTRIBUTING.md]: Yes, but I think having a direct route is better. No strong opinion though.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -16,3 +18,36 @@ Prometheus uses GitHub to manage reviews of pull requests.
...
+For quickly compiling and testing your changes do:
+```
+# For building.
+go build ./cmd/prometheus/
+./prometheus
+
+# For testing.
+make test-short   # To quickly test.
+
+make test         # For more thorough testing before you commit and push.
```
Comment (fabxc) [inline CONTRIBUTING.md]: Is it really worth distinguishing here?
While both exist, I think we never properly defined what makes a test short or long and running everything via `make test` just takes a few seconds. We should probably remove the short thing for now and only add it back once we have long running stuff like SD integration.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -16,3 +18,36 @@ Prometheus uses GitHub to manage reviews of pull requests.
...
+```
+
+All our issues are regularly tagged so that you can also filter down the issues involving the components you want to work on. For our labelling policy refer [the wiki page](https://github.com/prometheus/prometheus/wiki/Label-Names-and-Descriptions).
+
+## Pull Request Checklist
+
+* Branch from the master branch and, if needed, rebase to the current master branch before submitting your pull request. If it doesn't merge cleanly with master you may be asked to rebase your changes.
+
+* Commits should be as small as possible, while ensuring that each commit is correct independently (i.e., each commit should compile and pass tests).
```
Comment (fabxc) [inline CONTRIBUTING.md]: "as small as possible" could lead to undesired results. E.g. when a helper function is added for something I don't want it to be its own commit.
But likely we have more problems with too long commits. We can go with this but keep in mind to adjust it in case we get a bunch of 10+ commit PRs.

File: CONTRIBUTING.md (hunk 1 above)
Comment (fabxc) [inline CONTRIBUTING.md]: I tend to agree with Brian. This could go stale rather quickly. One hop to the community page shouldn't be too bad.

**Comment by fabxc** [comment]
Great stuff. Left a few comments.

File: CONTRIBUTING.md (hunk 2 above)
Comment (Gouthamve) [inline CONTRIBUTING.md]: Hmm, true this is more for 1.x where the storage tests take a little long.

Comment (Gouthamve) [inline CONTRIBUTING.md]: Fixed this.

**Comment by fabxc** [comment]
👍
